modelservice/master.py

```diff
@@ -90,8 +90,31 @@ def BuildModels(self, request: BuildModelsRequest, context) -> BuildModelsRespon
         )
 
     def GetModel(self, request: GetModelRequest, context) -> GetModelResponse:
-        info(f"Received request to retrieve model(s)")
-        return GetModelResponse()
+        info(f"Received request to retrieve model")
+
+        # job_id: str = request.model_id
+        gis_join: str = request.gis_joins
+
+        try:
+            worker: WorkerMetadata = self.gis_join_locations[gis_join]
+        except Exception as err:
+            print(err)
+            return GetModelResponse(
+                model_id=request.id,
+                error_occurred=True,
+                error_msg="Error retrieving requested GIS join",
+                filename="",
+                data=""
+            )
+
+        info(f"Launching get worker model for {worker.hostname}...")
+        with grpc.insecure_channel(f"{worker.hostname}:{worker.port}") as channel:
+            stub = modelservice_pb2_grpc.WorkerStub(channel)
+            # TODO: Determine if we need to make a copy when we just do a pass through
+            request_copy = GetModelRequest()
+            request_copy.CopyFrom(request)
```

Comment:
No need to copy request here you can use a reference, we're not changing anything in it

Reply:
Removed